ntoskrnl/cc/view.c

```diff
@@ -75,13 +75,16 @@ ULONG CcRosVacbIncRefCount_(PROS_VACB vacb, PCSTR file, INT line)
 ULONG CcRosVacbDecRefCount_(PROS_VACB vacb, PCSTR file, INT line)
 {
     ULONG Refs;
+    BOOLEAN VacbDirty = vacb->Dirty;
+    BOOLEAN VacbTrace = vacb->SharedCacheMap->Trace;
+    BOOLEAN VacbPageOut = vacb->PageOut;
```

Comment:
Are you certain that the compiler won't do a funny thing during optimization step and won't move, for example, `vacb->Dirty` at the place of usage of `VacbDirty` variable?

Reply:
No, Interlocked* is always a full memory barrier.